transcom/mymove
# [MB-16383] authz check for update progear endpoint

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16383)

## Summary

This PR adds an authz check for the PATCH `/ppm-shipments/{ppmShipmentId}/pro-gear-weight-tickets/{proGearWeightTicketId}` endpoint. It checks that the service member associated with the session matches the service member associated with the progear weight ticket

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Start server and customer client.
2. Log in as a service member that has an approved PPM shipment. I used `progearWeightTicket@ppm.approved`
3. Look at the network tab and find the values for the `PPMShipmentID` ,`ProgearWeightTicketID` and the eTag for progear
4. Log out and log in as another service member to get their cookie info for postman. Load into postman based on however you set up postman.
    1. [Docs for setting up postman](https://transcom.github.io/mymove-docs/docs/tools/postman/setting-up-postman)
5. Make a patch request in postman to `/mto-shipments` 
    1. Since this is going to be a patcht request, you'll need to [set the `X-CSRF-TOKEN` header to be the `masked_gorilla_csrf` cookie value](https://transcom.github.io/mymove-docs/docs/tools/postman/intercepting-cookies-for-postman#form-a-working-get-and-patch-request-in-postman).

6. Verify you get a 404 not found response.
  
### Screenshots
<img width="1398" alt="Screenshot 2023-08-10 at 4 38 25 PM" src="https://github.com/transcom/mymove/assets/67110378/650be0fa-2f98-4dca-885b-716284b049ac">
